shells/android/java/arcs/crdt/parcelables/ParcelableCrdtSingleton.kt

```diff
@@ -0,0 +1,136 @@
+package arcs.crdt.parcelables
+
+import android.os.Parcel
+import android.os.Parcelable
+import arcs.common.Referencable
+import arcs.common.ReferenceId
+import arcs.crdt.CrdtSet
+import arcs.crdt.CrdtSingleton
+
+object ParcelableCrdtSingleton {
+    data class Data(
+        override val actual: CrdtSingleton.Data<Referencable>
+    ) : ParcelableCrdtData<CrdtSingleton.Data<Referencable>> {
+        override var versionMap = actual.versionMap
+
+        override fun writeToParcel(parcel: Parcel, flags: Int) {
+            parcel.writeTypedObject(actual.versionMap.toParcelable(), flags)
+            parcel.writeInt(actual.values.size)
+            actual.values.forEach { (actor, value) ->
+                parcel.writeString(actor)
+                parcel.writeTypedObject(ParcelableCrdtSet.DataValue(value), flags)
+            }
+        }
+
+        override fun describeContents(): Int = 0
+
+        companion object CREATOR : Parcelable.Creator<Data> {
+            override fun createFromParcel(parcel: Parcel): Data {
+                val versionMap = requireNotNull(
+                    parcel.readTypedObject(ParcelableVersionMap.CREATOR)
+                ) { "No VersionMap found in parcel when reading ParcelableCrdtSingleton.Data" }
+
+                val values = mutableMapOf<ReferenceId, CrdtSet.DataValue<Referencable>>()
+                val items = parcel.readInt()
+                repeat(items) {
+                    values[requireNotNull(parcel.readString())] =
+                        requireNotNull(
+                            parcel.readTypedObject(ParcelableCrdtSet.DataValue.CREATOR)
+                        ) {
+                            "No DataValue found in parcel when reading ParcelableCrdtSingleton.Data"
+                        }.actual
```

Comment:
You can move the `actual` inside of the requireNotNull:

```kotlin
requireNotNull(parcel.readTypedObject(ParcelableCrdtSet.DataValue.CREATOR)?.actual) {
    "No DataValue found in parcel when reading ParcelableCrdtSingleton.Data"
}
```

Reply:
Done